espnet/bin/st_trans.py

```diff
@@ -0,0 +1,171 @@
+                        help='RNNLM weight')
+    # streaming related
+    parser.add_argument('--streaming-mode', type=str, default=None,
+                        choices=['window', 'segment'],
+                        help="""Use streaming translator for inference.
+                        `--batchsize` must be set to 0 to enable this mode""")
+    parser.add_argument('--streaming-window', type=int, default=10,
+                        help='Window size')
+    parser.add_argument('--streaming-min-blank-dur', type=int, default=10,
+                        help='Minimum blank duration threshold')
+    parser.add_argument('--streaming-onset-margin', type=int, default=1,
+                        help='Onset margin')
+    parser.add_argument('--streaming-offset-margin', type=int, default=1,
+                        help='Offset margin')
+    # multilingual related
+    parser.add_argument('--tgt-lang', default=False, type=str,
+                        help='target language ID (e.g., <en>, <de>, and <fr> etc.)')
+    return parser
+
+
+def main(args):
+    parser = get_parser()
+    args = parser.parse_args(args)
+
+    # logging info
+    if args.verbose == 1:
+        logging.basicConfig(
+            level=logging.INFO, format="%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s")
+    elif args.verbose == 2:
+        logging.basicConfig(level=logging.DEBUG,
+                            format="%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s")
+    else:
+        logging.basicConfig(
+            level=logging.WARN, format="%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s")
+        logging.warning("Skip DEBUG/INFO messages")
+
+    # check CUDA_VISIBLE_DEVICES
+    if args.ngpu > 0:
+        cvd = os.environ.get("CUDA_VISIBLE_DEVICES")
+        if cvd is None:
+            logging.warning("CUDA_VISIBLE_DEVICES is not set.")
+        elif args.ngpu != len(cvd.split(",")):
+            logging.error("#gpus is not matched with CUDA_VISIBLE_DEVICES.")
+            sys.exit(1)
+
+        # TODO(mn5k): support of multiple GPUs
+        if args.ngpu > 1:
+            logging.error("The program only supports ngpu=1.")
+            sys.exit(1)
+
+    # display PYTHONPATH
+    logging.info('python path = ' + os.environ.get('PYTHONPATH', '(None)'))
+
+    # seed setting
+    random.seed(args.seed)
+    np.random.seed(args.seed)
+    logging.info('set random seed = %d' % args.seed)
+
+    # validate rnn options
+    if args.rnnlm is not None and args.word_rnnlm is not None:
+        logging.error("It seems that both --rnnlm and --word-rnnlm are specified. Please use either option.")
+        sys.exit(1)
+
+    # trans
+    logging.info('backend = ' + args.backend)
+    if args.backend == "pytorch":
+        # Experimental API that supports custom LMs
+        if args.api == "v2":
+            from espnet.asr.pytorch_backend.recog import recog_v2
+            recog_v2(args)
```

Comment:
Can you add test for `espnet.nets.beam_search.BeamSearch` with ST models under test_beam_search?

Reply:
I will remove this for the clarity of this PR.

Reply:
OK!